autora/synthetic/inventory.py

```diff
@@ -115,33 +143,63 @@ class SyntheticExperimentCollection:
     experiment_runner: Optional[Callable] = None
     ground_truth: Optional[Callable] = None
     plotter: Optional[Callable[[Optional[_SupportsPredict]], None]] = None
+    closure: Optional[Callable] = None
 
 
-_Inventory: Dict[str, _SyntheticExperimentClosure] = dict()
+Inventory: Dict[str, _SyntheticExperimentClosure] = dict()
 """ The dictionary of `SyntheticExperimentCollection`. """
 
 
 def register(id_: str, closure: _SyntheticExperimentClosure) -> None:
     """
-    Add a new synthetic experiment to the _Inventory.
+    Add a new synthetic experiment to the Inventory.
 
     Parameters:
          id_: the unique id for the model.
          closure: a function which returns a SyntheticExperimentCollection
 
     """
-    _Inventory[id_] = closure
+    Inventory[id_] = closure
 
 
 def retrieve(id_: str, **kwargs) -> SyntheticExperimentCollection:
-    """Retrieve a synthetic experiment from the _Inventory.
+    """
+    Retrieve a synthetic experiment from the Inventory.
 
     Parameters:
         id_: the unique id for the model
         **kwargs: keyword arguments for the synthetic experiment (metadata, coefficients etc.)
     Returns:
         the synthetic experiment
     """
-    closure: _SyntheticExperimentClosure = _Inventory[id_]
+    closure: _SyntheticExperimentClosure = Inventory[id_]
     evaluated_closure = closure(**kwargs)
+    evaluated_closure.closure = closure
     return evaluated_closure
+
+
+@singledispatch
+def describe(arg):
+    """
+    Print the docstring for a synthetic experiment.
+
+    Args:
+        arg: the experiment's ID, an object returned from the `retrieve` function, or a closure
+            which creates a new experiment.
+    """
+    raise NotImplementedError(f"{arg=} not yet supported")
+
+
+@describe.register
+def _(closure: _SyntheticExperimentClosure):
+    print(closure.__doc__)
+
+
+@describe.register
+def _(collection: SyntheticExperimentCollection):
+    describe(collection.closure)
+
+
+@describe.register
+def _(id_: str):
+    describe(retrieve(id_))
```

Comment:
Very cool! Didn't know about this! Is my understanding correct that if the object type is not one of the of the registered types it will default to the base (first) function?

Reply:
Yes! That's it. It's just a different way of doing 
```python
if isinstance(x, SyntheticExperimentCollection): 
    ... do something ... 
elif isinstance(x, str):
    ... do something else ...
else:
    ... throw an error
```
which neatly leverages the type hints.

Reply:
Neat!